feat(Employee): update Job Applicant and Job Offer status on insert

krantheman opened the pull request:
Employee insert triggers updating of Job Applicant and Job Offer status as "Accepted", and subsequent submitting of Job Offer.

`no-docs`

krantheman committed:
feat(Employee): update job applicant status on insert

krantheman committed:
feat(Employee): update job offer status and submit

krantheman committed:
chore: move create_job_applicant to test_utils to prevent circular import

krantheman committed:
test: add test for update_job_applicant_and_offer

ruchamahabal commented on hrms/overrides/employee_master.py:
```diff
@@ -65,6 +65,16 @@ def update_employee_transfer(doc, method=None):
+	if doc.job_applicant:
+		frappe.db.set_value("Job Applicant", doc.job_applicant, "status", "Accepted")
+		job_offer = frappe.get_last_doc("Job Offer", filters={"job_applicant": doc.job_applicant})
+		job_offer.status = "Accepted"
```
```suggestion
		job_offer.status = "Accepted"
		job_offer.flags.ignore_mandatory = True
		job_offer.flags.ignore_permissions = True
```

If someone has customized the Job Offer doctype to make some field mandatory or doesn't have permission to update it, this will error out. Since its a backend operation, user will get confused in case of permission/mandatory errors from another doctype, so lets do it with these flags

ruchamahabal replied:
Also inform the user about this implicit action with a msgprint

ruchamahabal commented on hrms/tests/test_utils.py:
```diff
@@ -92,3 +93,25 @@ def create_department(name: str, company: str = "_Test Company") -> str:
 	return department.name
+
+
+def create_job_applicant(**args):
```
+1 for creating this util

krantheman replied:
Um, actually I didn't create it; I just moved it from `test_job_applicant.py` 😅

krantheman committed:
fix: permission/mandatory errors

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

krantheman committed:
feat: add msgprint for changing Job Applicant/Offer status

ruchamahabal committed:
refactor: make auto-update message more informative

- place job applicant update/validate functions together

- replace implicit submission for job offer with a message

ruchamahabal commented:
Unrelated test failure. Will fix separately
